src/gluon/services/team/AddMemberToTeamService.ts

```
@@ -13,15 +17,19 @@ export class AddMemberToTeamService {
     constructor(private gluonService = new GluonService()) {
     }
 
-    public async getNewMember(chatId: string, teamChannel: string) {
-        const newMember = await this.gluonService.members.gluonMemberFromScreenName(chatId);
+    public async getNewMember(ctx: HandlerContext, chatId: string, teamChannel: string) {
+        try {
+            const newMember = await this.gluonService.members.gluonMemberFromScreenName(chatId);
 
-        if (!_.isEmpty(_.find(newMember.teams,
-            (team: any) => team.slack.teamChannel === teamChannel))) {
-            throw new QMError(`${newMember.slack.screenName} is already a member of this team.`);
+            return this.partOfTeam(newMember, teamChannel);
+        } catch (error) {
+            if (error.message === `${chatId} is already a member of this team.`) {
+                throw new QMError(`${chatId} is already a member of this team.`);
```

I think you can maybe just re-throw the error here. Also maybe catch and re-throw errors that aren't QM Errors I.E:
```javascript
catch(error){
  const isQMError = error instanceof QMError;
  if (!isQMError || (isQMError && error.message === `${chatId} is already a member of this team.`)) {
        throw error;
  }
  //... carry on
}
```